internal/go_repository.bzl

```diff
@@ -117,6 +117,51 @@ Authorization: Bearer RANDOM-TOKEN
 # We can't disable timeouts on Bazel, but we can set them to large values.
 _GO_REPOSITORY_TIMEOUT = 86400
 
+def _escape_go_module_path(module_path):
+    """Escapes a Go module path according to the Go proxy protocol."""
+    result = ""
+    for c in module_path.elems():
+        if "A" <= c and c <= "Z":
+            result += "!" + c.lower()
+        else:
+            result += c
+    return result
+
+def _get_goproxy_url_for_module(env, module_path):
+    """Return proxy URL to be used for module_path, or None if no proxy will be used."""
+    goproxy = env.get("GOPROXY", "")
+    goprivate = env.get("GOPRIVATE", "")
+    gonoproxy = env.get("GONOPROXY", "")
+
+    if goproxy == "off":
+        return None
+
+    def matches_pattern(pattern, module_path):
+        pat = pattern.strip()
+        if not pat:
+            return False
+        if pat.endswith("/..."):
+            return module_path.startswith(pat[:-4])
```

Comment:
This would let `foo/bar/...` match `foo/barbaz`.

Reply:
actually i'll remove this for now, given https://github.com/bazel-contrib/bazel-gazelle/pull/2124#discussion_r2160555840